puppet/modules/jenkins_job_builder/manifests/init.pp

```diff
@@ -0,0 +1,60 @@
+# Class to install, configure, and maintain JJB, as well
+# as to deploy the actual jobs to jenkins
+#
+# Loosely based on
+# https://git.openstack.org/cgit/openstack-infra/puppet-jenkins/tree/manifests/job_builder.pp
+# and should probably be kept up to date from there
+#
+class jenkins_job_builder (
+  $url = '',
+  $username = '',
+  $password = '',
+  $config_dir = '',
+) {
+
+  if ! defined(Package['jenkins-job-builder']) {
+    package { 'jenkins-job-builder':
+      ensure   => present,
+      provider => 'pip',
+    }
+  }
+
+  file { '/etc/jenkins_jobs':
+    ensure => directory,
+  }
+
+  file { '/etc/jenkins_jobs/config':
+    ensure  => directory,
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0755',
+    recurse => true,
+    purge   => true,
+    force   => true,
+    source  => $config_dir,
+    require => File['/etc/jenkins_jobs'],
+    notify  => Exec['jenkins_jobs_update'],
+  }
+
+  exec { 'jenkins_jobs_update':
+    #command    => 'jenkins-jobs update --delete-old /etc/jenkins_jobs/config', #danger!
+    #command    => 'jenkins-jobs update /etc/jenkins_jobs/config',              #op
```

Comment:
How about parameterising this?  Then we could enable/disable it from Foreman, or even have a staging-jenkins server which is always up to date or something.